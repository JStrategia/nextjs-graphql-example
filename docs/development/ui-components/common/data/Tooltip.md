# Tooltip

Wraps children in a simple tooltip container. On hover, tooltip is displayed under the wrapped element.

## Usage

```tsx
function MyComponent() {
  return (
    <Tooltip title="Some text that will be displayed on hover">
      <Body1>Some children</Body1>
    </Tooltip>
  );
}
```
