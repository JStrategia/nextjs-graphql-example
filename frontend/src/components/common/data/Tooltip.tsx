import React, { ReactElement } from 'react';
import styled, { css } from 'styled-components';
import Typography from 'components/common/typography/Typography';

type TooltipProps = {
  readonly children: ReactElement;
  readonly title: string;
};

function Tooltip({ children, title }: TooltipProps) {
  return (
    <TooltipContainer>
      {children}
      <TooltipText data-testid="tooltip-text">{title}</TooltipText>
    </TooltipContainer>
  );
}

const TooltipText = styled.span(
  ({ theme }) => css`
    ${Typography};
    visibility: hidden;
    min-width: ${theme.spacing.tooltip};
    background-color: ${theme.colors.tooltip.background};
    color: ${theme.colors.tooltip.onBackground};
    font-size: ${theme.fonts.sizes.xs};
    text-align: center;
    padding: ${theme.spacing.xxs};
    border-radius: ${theme.spacing.xxs};
    margin-top: ${theme.spacing.xs};
    transform: translateX(-50%);
    left: 50%;

    position: absolute;
    z-index: 1;
  `,
);

const TooltipContainer = styled.div`
  position: relative;
  display: inline-block;

  &:hover ${TooltipText} {
    visibility: visible;
  }
`;

export default Tooltip;
export type { TooltipProps };
