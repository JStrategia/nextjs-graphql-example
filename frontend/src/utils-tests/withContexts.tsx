import React from 'react';
import ThemeProvider from 'contexts/ThemeProvider';

function withContexts(children: React.ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
