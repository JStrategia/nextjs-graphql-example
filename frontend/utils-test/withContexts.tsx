import React, { ReactNode } from 'react';
import ThemeProvider from '../contexts/ThemeProvider';

function withContexts(children: ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
