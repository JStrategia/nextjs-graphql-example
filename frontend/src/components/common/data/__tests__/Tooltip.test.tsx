import React from 'react';
import { create } from 'react-test-renderer';
import Tooltip from 'components/common/data/Tooltip';
import withContexts from 'utils-tests/withContexts';

describe('Tooltip', () => {
  describe('Given title and children', () => {
    const title = 'some title';
    const children = <span>Children</span>;

    describe('When rendering', () => {
      const render = create(withContexts(<Tooltip title={title}>{children}</Tooltip>)).toJSON();

      test('Then match snapshot', () => {
        expect(render).toMatchSnapshot();
      });
    });
  });
});
