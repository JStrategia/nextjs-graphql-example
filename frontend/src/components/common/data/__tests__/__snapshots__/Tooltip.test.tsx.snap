// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Given title and children When rendering Then match snapshot 1`] = `
<div
  className="sc-gsnTZi eZrxQK"
>
  <span>
    Children
  </span>
  <span
    className="sc-bczRLJ gnxjFk"
  >
    some title
  </span>
</div>
`;
